components/Habit.js

```diff
@@ -4,19 +4,30 @@ import InlineEdit from './InlineEdit';
 
 export default function Habit(props) {
 
+  const {setCompletedHabits, completedHabits } = props;
+
   const [editingValue, setEditingValue] = useState(props.habit.text);
   const [checkmark, setCheckmark] = useState('/grayCheckMark.svg');
   const [isEmojiModalShowing, setIsEmojiModalShowing] = useState(false);
   const [emoji, setEmoji] = useState(props.habit.emoji);
 
-  const deleteHandler = () => props.deleteHandler(props.id);
-
-  const styleCheckmark = () => {
+  const deleteHandler = (e) => {
+    e.stopPropagation()
+    props.deleteHandler(props.id);
+  }
+  const selectCard = () => {
     if (checkmark === "/redCheckMark.svg") {
       setCheckmark("/grayCheckMark.svg");
     } else {
       setCheckmark("/redCheckMark.svg");
     }
+    if (props.habit.selected === false) {
+      setCompletedHabits(completedHabits => completedHabits + 1)
+      props.habit.selected = true;
```

Comment:
Updating a prop value within the component is a React antipattern. React props are meant to be **immutable** values passed in by the parent into a child, so its not meant to be updated within a React component like this.